### PR TITLE
chore: add more required checks

### PR DESCRIPTION
**Pull request by eschutho**
### SUMMARY
Continuation of https://github.com/apache/superset/pull/12921, this PR adds more required github checks. 

### TEST PLAN
verify that the correct tests are required after this is merged. If this yml file doesn't match correctly, we will need to rename the checks so that they match or open an infra ticket to Apache to update it.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by eschutho** [comment]
@ktmud @nytai

File: .asf.yaml [hunk 1]
```diff
@@ -61,6 +61,8 @@ github:
         # contexts are the names of checks that must pass
         contexts:
           - check
+          - lint (3.7)
```
Comment (ktmud) [inline .asf.yaml:64]: Do you think it's worth updating `lint` to `python-lint` as well?

**Review by ktmud — APPROVED** [review]

**Comment by ktmud** [comment]
With the flexibility to set our own required checks, I think it's now time to get rid off the no-op ["E2E / Cypress (chrome)"](https://github.com/apache/superset/blob/1f27b62d51c37f6ed28d6f446df9085d8a0ede4e/.github/workflows/superset-e2e.yml#L114) job that was obviously only meant to appease the original required checks.

**Comment by robdiciuccio** [comment]
@ktmud yes, the `E2E / Cypress` job was purely to satisfy the required checks. Is the plan to remove e2e as a required check? If not, it may still be needed due to how the matrix operates.

**Comment by ktmud** [comment]
> @ktmud yes, the `E2E / Cypress` job was purely to satisfy the required checks. Is the plan to remove e2e as a required check? If not, it may still be needed due to how the matrix operates.

I think we do want to keep them as required checks. We can remove the no-op job and add required checks on `cypress-matrix (1, chrome)`, `cypress-matrix (2, chrome)` and `cypress-matrix (3, chrome)` instead.

File: .asf.yaml (hunk 1 above)
Comment (eschutho) [reply .asf.yaml]: yes, good idea. I removed it from this PR, added test-mysql instead. I'll rename lint in a separate pr

**Comment by eschutho** [comment]
@ktmud do you think we should merge this on a Friday? I can set aside time to test, since this is the first time we're adding a check with the matrix.

**Comment by eschutho** [comment]
🏷 ready-to-merge

**Pull request merged**